### ISSUE TITLE
Add tray-icon support

### ISSUE THREAD

**Issue by qtpies** [paolostivanin/OTPClient#306]
Copied from https://gitlab.gnome.org/World/Authenticator/-/issues/386. Asking here, as Authenticator is apparently exclusively developing for gnome desktop...  

otpclient is an application that is needed a lot, at random moments. I think it would be justified for otpclient to be an 'always on' application that lives in the system tray next to network manager, sound settings etc. This way I could get a 2FA code with two clicks, and after that the otpclient window disappears with no action required when I click in the field where I enter the code. This is better than having the otpclient window floating around when not in use. It would be a nice enhancement of the Linux desktop experience to have this option.

Would you consider putting this on the roadmap somewhere?

**Comment by paolostivanin** [comment]
Hello,
to be honest, I've thought about it a lot during these years. I've never added this feature because, while OTPClient works on every DE, it is developed and tested only on GNOME.
Supporting all DEs is unfeasible because, unfortunately, OTPClient is a one-man-show. Therefore I have to carefully prioritize what to develop, because it also means that I have to maintain it too. If I start adding features that are not supported on GNOME, it means that I have to spend even more time on testing, because I'd have to check others DEs too.
Moreover, with more and more distro (and DEs) planning to go Wayland only, this feature wouldn't last much since Wayland doesn't support tray icon (this is an X11 only feature).

TL;DR: sorry, I don't have the capacity to implement this feature. I'll leave this open for the time being as "contributor-only".

**Comment by qtpies** [comment]
I totally understand. Does anyone have further info though about wayland not support tray icons? Is wayland trying to force one style of desktop usage?

- I have looked around a lot, but all I find are bug reports from apps minimizing to tray. I can't find an explanation by Wayland developers.
- GNOME on wayland still has tray icons for WiFi, sound an battery, like on [this](https://www.reddit.com/media?url=https%3A%2F%2Fexternal-preview.redd.it%2FaWlC6mOV-CFiiEVDujeLAQKk11-bJU2qNqp-_uYNLb4.jpg%3Fauto%3Dwebp%26s%3D0833dbd77aa0c0d98b1d9fa5f2d2af25852d685e) screenshot. Why can't 'third party' applications have a tray icon then?  
- [waybar](https://github.com/Alexays/Waybar) which is specifically made for wayland actually has a lot of systray icons...

**Comment by paolostivanin** [comment]
Those you see on GNOME are not tray icons. It's something integrated in the DE itself, it's part of gnome-shell.
About waybar, I have no idea what it is and how it works :smile: I'm still on X11 due to nVidia and multi screens.

**Comment by paolostivanin** [comment]
@qtpies waybar is only for **sway** and **Wlroots** based compositors, which means that swaybar won't work on GNOME and KDE desktops, for example.

**Comment by paolostivanin** [comment]
closing, as this will never be implemented.
